azurerm/internal/services/servicebus/validate/subscription_name.go

```diff
@@ -0,0 +1,15 @@
+package validate
+
+import (
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func SubscriptionName() schema.SchemaValidateFunc {
+	return validation.StringMatch(
+		regexp.MustCompile("^[a-zA-Z0-9][-._a-zA-Z0-9]{0,48}([a-zA-Z0-9])?$"),
+		"The name can contain only letters, numbers, periods, hyphens and underscores. The name must start and end with a letter or number and be less then 50 characters long.",
```

Comment:
This regexp will allow up to 50 chars, but the error message specifies less than 50?

Reply:
48 + the start and end, so this is 50 :)